### PR TITLE
fix: Set Deal Status Completed

### PR DESCRIPTION
**Pull request by inchori**
Set deal status `COMPLETED` when deal's `cur_num_data` and `max_num_data` is equal.

### CODE REVIEW

File: x/market/keeper/deal_test.go [hunk 1]
```diff
@@ -516,5 +552,29 @@ func makeTestCert() types.DataValidationCertificate {
...
 	}
+}
+
+func makeTestDataCert2() types.DataValidationCertificate {
+	uCert := types.UnsignedDataValidationCertificate{
+		DealId:               2,
+		DataHash:             []byte("1a312c1223x2fs3"),
+		EncryptedDataUrl:     []byte("https://panacea.org/a/123.json"),
+		DataValidatorAddress: newAddr.String(),
+		RequesterAddress:     acc2.String(),
```
Comment (gyuguen) [inline x/market/keeper/deal_test.go:563]: It seems to me that only this address is a different value.
How about it takes an address as a parameter?

Comment (inchori) [reply x/market/keeper/deal_test.go:563]: Ah you means the function likes ```func makeTestDataCert(address sdk.AccAddress) { ... }``` ?

Comment (gyuguen) [reply x/market/keeper/deal_test.go:563]: That's right.

**Review by hansol-medi — APPROVED** [review]
LGTM.

The fail case of sell data to `COMPLETED` deal is treated in `TestDealIsNotActive()`.
In `TestDealIsNotActive()`, the status of deal is set directly, not by actual sell of data.
For better test of `COMPLETED` of deal, I think it's better to sell max num of data to the deal (to make deal status `COMPLETED`) and try sell more data with assertion of error.
This scenario can actually check if the deal set `COMPLETED` when max num of data sold.

File: x/market/keeper/deal_test.go (hunk 1 above)
Comment (inchori) [reply x/market/keeper/deal_test.go:563]: Not only different for address, but also for ```dataHash```. So I will takes a ```dataHash``` and ```address``` as a parameter.

Comment (gyuguen) [reply x/market/keeper/deal_test.go:563]: Thank you :)

**Review by gyuguen — APPROVED** [review]
LGTM 👍

**Review by youngjoon-lee — COMMENTED** [review]
LGTM

**Pull request merged**